pkg/gardenlet/controller/seed/seed.go

```diff
@@ -189,48 +188,25 @@ func (c *Controller) Run(ctx context.Context, workers int) {
 }
 
 func (c *Controller) startHealthManagement(ctx context.Context) {
-	var (
-		seedName              = confighelper.SeedNameFromSeedConfig(c.config.SeedConfig)
-		seedLabelSelector     labels.Selector
-		expectedHealthReports int
-		err                   error
-	)
-
-	if seedName != "" {
-		expectedHealthReports = 1
-	} else {
-		seedLabelSelector, err = metav1.LabelSelectorAsSelector(c.config.SeedSelector)
-		if err != nil {
-			panic(err)
-		}
-	}
+	seedName := confighelper.SeedNameFromSeedConfig(c.config.SeedConfig)
+	expectedHealthReports := 1 // TODO: Shouldn't this be reset for each loop iteration?
```

Comment:
Hmm, I think in https://github.com/gardener/gardener/pull/2925 the goal was to not let the health checks fail once the seed got deleted to avoid unnecessary restarts of the gardenlet and unwanted re-registrations of the seed.
So, I think it's fine like this.

@rfranzke do you remember more details / any additional thoughts?

Reply:
Not really, sorry.